### PR TITLE
Make force computations use mutable slices.

### LINKED ISSUE #172: Forces computations should not allocate a force `Vec` and return it
They should instead take a mutable slice as a parameter. The `Vec` always already exists, so by doing so we:

- use extra memory
- have to manually sum the `Vec` after the calls like [here](https://github.com/lumol-org/lumol/blob/8b410a0457258751b7deb7fbb07c8a8926283230/src/core/src/sys/compute.rs#L102-L104), which is kind of weird

### PR DESCRIPTION
**Pull request by antoinewdg**
Fix #172

### CODE REVIEW

**Comment by antoinewdg** [comment]
(I labelled it as performance related because it deserves a benchmark to check if it does not impact it)

File: src/core/src/energy/global/ewald.rs [hunk 1]
```diff
@@ -759,15 +759,13 @@ impl GlobalPotential for SharedEwald {
         return real + self_e + kspace + molecular;
     }
 
-    fn forces(&self, system: &System) -> Vec<Vector3D> {
+    fn forces(&self, system: &System, forces: &mut [Vector3D])  {
```
Comment (Luthaf) [inline src/core/src/energy/global/ewald.rs]: Be cause we no longer allocate the data ourself, we will need to assert that the slice has the right length whenever we get one as a parameter.

Comment (antoinewdg) [inline src/core/src/energy/global/ewald.rs]: Done

File: benches/water.rs [hunk 2]
```diff
@@ -41,9 +41,10 @@ fn energy_ewald(bencher: &mut Bencher) {
 fn forces_ewald(bencher: &mut Bencher) {
     let system = utils::get_system("water");
     let ewald = get_ewald();
+    let mut forces = vec![Vector3D::zero(); system.size()];
```
Comment (antoinewdg) [inline benches/water.rs]: I put the forces initialization out of the loop to avoid benchmarking it, is that a correct way to do it ?

Comment (Luthaf) [inline benches/water.rs]: Yep, that's right. The code will be executed each time, but not taken in account for the loop timing.

**Comment by Luthaf** [comment]
## Comparing to master (8b410a04)
using `--threshold 2, latest commit first`
<details><summary>2098cc4d Adapt benchmarks to new forces signature</summary>
  <p>





```bash

 name                        8b410a04.txt ns/iter    2098cc4d.txt ns/iter     diff ns/iter  diff % 
 argon::cache_move_particle  334,679 (+/- 2596)      325,272 (+/- 3124)             -9,407  -2.81% 
 nacl::forces_ewald          25,329,807 (+/- 98650)  23,486,274 (+/- 124249)    -1,843,533  -7.28% 
 nacl::forces_wolf           156,620 (+/- 1874)      149,910 (+/- 1465)             -6,710  -4.28% 
 water::forces_ewald         37,163,464 (+/- 96410)  34,766,397 (+/- 77494)     -2,397,067  -6.45% 
 water::forces_wolf          205,120 (+/- 1958)      223,300 (+/- 5009)             18,180   8.86% 



```



</p></details>

<details><summary>01e563e4 Assert that force slices have a correct size</summary>
  <p>





```bash




```



</p></details>

<details><summary>b2e8aebb Make force computations use mutable slices.</summary>
  <p>





```bash




```



</p></details>

## Individual benchmarks

<details><summary>2098cc4d Adapt benchmarks to new forces signature</summary>
  <p>





```bash


running 4 tests
test argon::cache_move_particle ... bench:     325,272 ns/iter (+/- 3,124)
test argon::energy              ... bench:     311,341 ns/iter (+/- 9,035)
test argon::forces              ... bench:     393,777 ns/iter (+/- 2,943)
test argon::virial              ... bench:     405,357 ns/iter (+/- 18,914)

test result: ok. 0 passed; 0 failed; 0 ignored; 4 measured


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured


running 8 tests
test nacl::cache_move_particle_ewald ... bench:     656,379 ns/iter (+/- 1,142)
test nacl::cache_move_particle_wolf  ... bench:     167,249 ns/iter (+/- 1,924)
test nacl::energy_ewald              ... bench:   1,156,926 ns/iter (+/- 982)
test nacl::energy_wolf               ... bench:      98,568 ns/iter (+/- 1,157)
test nacl::forces_ewald              ... bench:  23,486,274 ns/iter (+/- 124,249)
test nacl::forces_wolf               ... bench:     149,910 ns/iter (+/- 1,465)
test nacl::virial_ewald              ... bench:   8,988,508 ns/iter (+/- 1,445,237)
test nacl::virial_wolf               ... bench:     113,126 ns/iter (+/- 738)

test result: ok. 0 passed; 0 failed; 0 ignored; 8 measured


running 5 tests
test propane::cache_move_all_rigid_molecules ... bench:   1,991,109 ns/iter (+/- 16,571)
test propane::cache_move_particles           ... bench:     509,365 ns/iter (+/- 3,504)
test propane::energy                         ... bench:     383,487 ns/iter (+/- 41,952)
test propane::forces                         ... bench:     477,457 ns/iter (+/- 34,366)
test propane::virial                         ... bench:     232,568 ns/iter (+/- 41,224)

test result: ok. 0 passed; 0 failed; 0 ignored; 5 measured


running 10 tests
test water::cache_move_all_rigid_molecules_ewald ... bench:  13,060,074 ns/iter (+/- 14,377)
test water::cache_move_all_rigid_molecules_wolf  ... bench:  11,613,945 ns/iter (+/- 385,356)
test water::cache_move_particles_ewald           ... bench:   1,369,167 ns/iter (+/- 7,694)
test water::cache_move_particles_wolf            ... bench:     761,968 ns/iter (+/- 30,129)
test water::energy_ewald                         ... bench:   1,877,285 ns/iter (+/- 1,759)
test water::energy_wolf                          ... bench:     137,325 ns/iter (+/- 11,913)
test water::forces_ewald                         ... bench:  34,766,397 ns/iter (+/- 77,494)
test water::forces_wolf                          ... bench:     223,300 ns/iter (+/- 5,009)
test water::virial_ewald                         ... bench:  13,076,617 ns/iter (+/- 961,870)
test water::virial_wolf                          ... bench:     163,269 ns/iter (+/- 19,833)

test result: ok. 0 passed; 0 failed; 0 ignored; 10 measured




```



</p></details>

<details><summary>01e563e4 Assert that force slices have a correct size</summary>
  <p>





```bash

Build failed, waiting for other jobs to finish...



```



</p></details>

<details><summary>b2e8aebb Make force computations use mutable slices.</summary>
  <p>





```bash

Build failed, waiting for other jobs to finish...
Build failed, waiting for other jobs to finish...



```



</p></details>

<details><summary>8b410a04 Parallelize Virial::compute</summary>
  <p>





```bash


running 4 tests
test argon::cache_move_particle ... bench:     334,679 ns/iter (+/- 2,596)
test argon::energy              ... bench:     312,629 ns/iter (+/- 27,096)
test argon::forces              ... bench:     394,416 ns/iter (+/- 2,557)
test argon::virial              ... bench:     404,210 ns/iter (+/- 4,731)

test result: ok. 0 passed; 0 failed; 0 ignored; 4 measured


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured


running 8 tests
test nacl::cache_move_particle_ewald ... bench:     653,821 ns/iter (+/- 1,255)
test nacl::cache_move_particle_wolf  ... bench:     165,690 ns/iter (+/- 2,556)
test nacl::energy_ewald              ... bench:   1,156,629 ns/iter (+/- 1,764)
test nacl::energy_wolf               ... bench:      98,577 ns/iter (+/- 4,327)
test nacl::forces_ewald              ... bench:  25,329,807 ns/iter (+/- 98,650)
test nacl::forces_wolf               ... bench:     156,620 ns/iter (+/- 1,874)
test nacl::virial_ewald              ... bench:   9,030,621 ns/iter (+/- 1,475,541)
test nacl::virial_wolf               ... bench:     112,270 ns/iter (+/- 1,013)

test result: ok. 0 passed; 0 failed; 0 ignored; 8 measured


running 5 tests
test propane::cache_move_all_rigid_molecules ... bench:   1,998,214 ns/iter (+/- 4,383)
test propane::cache_move_particles           ... bench:     516,167 ns/iter (+/- 2,712)
test propane::energy                         ... bench:     381,136 ns/iter (+/- 25,252)
test propane::forces                         ... bench:     477,951 ns/iter (+/- 38,090)
test propane::virial                         ... bench:     231,687 ns/iter (+/- 4,376)

test result: ok. 0 passed; 0 failed; 0 ignored; 5 measured


running 10 tests
test water::cache_move_all_rigid_molecules_ewald ... bench:  13,167,243 ns/iter (+/- 15,566)
test water::cache_move_all_rigid_molecules_wolf  ... bench:  11,750,079 ns/iter (+/- 765,680)
test water::cache_move_particles_ewald           ... bench:   1,394,471 ns/iter (+/- 38,217)
test water::cache_move_particles_wolf            ... bench:     766,784 ns/iter (+/- 41,748)
test water::energy_ewald                         ... bench:   1,890,362 ns/iter (+/- 1,585)
test water::energy_wolf                          ... bench:     137,549 ns/iter (+/- 7,514)
test water::forces_ewald                         ... bench:  37,163,464 ns/iter (+/- 96,410)
test water::forces_wolf                          ... bench:     205,120 ns/iter (+/- 1,958)
test water::virial_ewald                         ... bench:  13,022,894 ns/iter (+/- 628,076)
test water::virial_wolf                          ... bench:     164,319 ns/iter (+/- 11,707)

test result: ok. 0 passed; 0 failed; 0 ignored; 10 measured




```



</p></details>

<br>**CPU**: Intel(R) Xeon(R) CPU E5-2660 0 @ 2.20GHz

**Comment by Luthaf** [comment]
I can not run the benchmarks for this PR, because they do not compile for intermediate commits. Could you squash as needed here?

**Comment by antoinewdg** [comment]
Done

**Comment by antoinewdg** [comment]
## Comparing to master (3d72dc1)
using `--threshold 2, latest commit first`
<details><summary>a4a8df4 Make force computations use mutable slices.</summary>
  <p>





```bash

 name                                         3d72dc1.txt ns/iter     a4a8df4.txt ns/iter     diff ns/iter   diff % 
 argon::energy                                450,972 (+/- 17989)     434,522 (+/- 23004)          -16,450   -3.65% 
 argon::virial                                610,204 (+/- 44730)     583,253 (+/- 44378)          -26,951   -4.42% 
 nacl::forces_ewald                           17,231,452 (+/- 47960)  14,463,171 (+/- 70247)    -2,768,281  -16.07% 
 propane::forces                              451,690 (+/- 27713)     468,095 (+/- 28748)           16,405    3.63% 
 water::cache_move_all_rigid_molecules_ewald  7,284,003 (+/- 77905)   7,443,936 (+/- 58189)        159,933    2.20% 
 water::forces_ewald                          24,931,876 (+/- 82610)  20,803,296 (+/- 94716)    -4,128,580  -16.56% 



```



</p></details>

## Individual benchmarks

<details><summary>a4a8df4 Make force computations use mutable slices.</summary>
  <p>





```bash


running 4 tests
test argon::cache_move_particle ... bench:     241,549 ns/iter (+/- 2,790)
test argon::energy              ... bench:     434,522 ns/iter (+/- 23,004)
test argon::forces              ... bench:     521,321 ns/iter (+/- 21,093)
test argon::virial              ... bench:     583,253 ns/iter (+/- 44,378)

test result: ok. 0 passed; 0 failed; 0 ignored; 4 measured


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured


running 8 tests
test nacl::cache_move_particle_ewald ... bench:     395,193 ns/iter (+/- 2,741)
test nacl::cache_move_particle_wolf  ... bench:      93,837 ns/iter (+/- 23,714)
test nacl::energy_ewald              ... bench:     620,803 ns/iter (+/- 2,567)
test nacl::energy_wolf               ... bench:     114,956 ns/iter (+/- 10,101)
test nacl::forces_ewald              ... bench:  14,463,171 ns/iter (+/- 70,247)
test nacl::forces_wolf               ... bench:     150,502 ns/iter (+/- 2,756)
test nacl::virial_ewald              ... bench:  13,886,334 ns/iter (+/- 465,402)
test nacl::virial_wolf               ... bench:     156,620 ns/iter (+/- 9,413)

test result: ok. 0 passed; 0 failed; 0 ignored; 8 measured


running 5 tests
test propane::cache_move_all_rigid_molecules ... bench:   1,035,425 ns/iter (+/- 13,421)
test propane::cache_move_particles           ... bench:     281,004 ns/iter (+/- 3,892)
test propane::energy                         ... bench:     391,885 ns/iter (+/- 14,446)
test propane::forces                         ... bench:     468,095 ns/iter (+/- 28,748)
test propane::virial                         ... bench:     348,601 ns/iter (+/- 27,025)

test result: ok. 0 passed; 0 failed; 0 ignored; 5 measured


running 10 tests
test water::cache_move_all_rigid_molecules_ewald ... bench:   7,443,936 ns/iter (+/- 58,189)
test water::cache_move_all_rigid_molecules_wolf  ... bench:   6,631,609 ns/iter (+/- 1,057,432)
test water::cache_move_particles_ewald           ... bench:     791,073 ns/iter (+/- 2,634)
test water::cache_move_particles_wolf            ... bench:     419,686 ns/iter (+/- 76,215)
test water::energy_ewald                         ... bench:   1,010,630 ns/iter (+/- 4,141)
test water::energy_wolf                          ... bench:     165,485 ns/iter (+/- 11,327)
test water::forces_ewald                         ... bench:  20,803,296 ns/iter (+/- 94,716)
test water::forces_wolf                          ... bench:     218,230 ns/iter (+/- 15,847)
test water::virial_ewald                         ... bench:  20,376,303 ns/iter (+/- 804,736)
test water::virial_wolf                          ... bench:     235,930 ns/iter (+/- 23,843)

test result: ok. 0 passed; 0 failed; 0 ignored; 10 measured




```



</p></details>

<details><summary>3d72dc1 Typo Orthorombic => Orthorhombic</summary>
  <p>





```bash


running 4 tests
test argon::cache_move_particle ... bench:     239,765 ns/iter (+/- 3,966)
test argon::energy              ... bench:     450,972 ns/iter (+/- 17,989)
test argon::forces              ... bench:     526,932 ns/iter (+/- 34,821)
test argon::virial              ... bench:     610,204 ns/iter (+/- 44,730)

test result: ok. 0 passed; 0 failed; 0 ignored; 4 measured


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured


running 8 tests
test nacl::cache_move_particle_ewald ... bench:     394,748 ns/iter (+/- 5,143)
test nacl::cache_move_particle_wolf  ... bench:      92,841 ns/iter (+/- 17,822)
test nacl::energy_ewald              ... bench:     617,578 ns/iter (+/- 2,304)
test nacl::energy_wolf               ... bench:     117,058 ns/iter (+/- 4,509)
test nacl::forces_ewald              ... bench:  17,231,452 ns/iter (+/- 47,960)
test nacl::forces_wolf               ... bench:     150,668 ns/iter (+/- 4,410)
test nacl::virial_ewald              ... bench:  13,798,847 ns/iter (+/- 529,885)
test nacl::virial_wolf               ... bench:     157,690 ns/iter (+/- 5,215)

test result: ok. 0 passed; 0 failed; 0 ignored; 8 measured


running 5 tests
test propane::cache_move_all_rigid_molecules ... bench:   1,027,341 ns/iter (+/- 21,346)
test propane::cache_move_particles           ... bench:     279,211 ns/iter (+/- 8,877)
test propane::energy                         ... bench:     391,686 ns/iter (+/- 19,789)
test propane::forces                         ... bench:     451,690 ns/iter (+/- 27,713)
test propane::virial                         ... bench:     351,999 ns/iter (+/- 20,898)

test result: ok. 0 passed; 0 failed; 0 ignored; 5 measured


running 10 tests
test water::cache_move_all_rigid_molecules_ewald ... bench:   7,284,003 ns/iter (+/- 77,905)
test water::cache_move_all_rigid_molecules_wolf  ... bench:   6,669,163 ns/iter (+/- 1,011,993)
test water::cache_move_particles_ewald           ... bench:     788,432 ns/iter (+/- 1,804)
test water::cache_move_particles_wolf            ... bench:     419,388 ns/iter (+/- 118,551)
test water::energy_ewald                         ... bench:   1,013,781 ns/iter (+/- 2,944)
test water::energy_wolf                          ... bench:     166,106 ns/iter (+/- 10,998)
test water::forces_ewald                         ... bench:  24,931,876 ns/iter (+/- 82,610)
test water::forces_wolf                          ... bench:     221,223 ns/iter (+/- 18,925)
test water::virial_ewald                         ... bench:  20,538,970 ns/iter (+/- 796,610)
test water::virial_wolf                          ... bench:     237,975 ns/iter (+/- 16,218)

test result: ok. 0 passed; 0 failed; 0 ignored; 10 measured




```



</p></details>

<br>**CPU**: Intel(R) Core(TM) i7-5700HQ CPU @ 2.70GHz

**Comment by Luthaf** [comment]
## Comparing to master (3d72dc18)
using `--threshold 2, latest commit first`
<details><summary>a4a8df4c Make force computations use mutable slices.</summary>
  <p>





```bash

 name                 3d72dc18.txt ns/iter    a4a8df4c.txt ns/iter     diff ns/iter  diff % 
 nacl::forces_ewald   25,330,907 (+/- 92705)  23,481,689 (+/- 116110)    -1,849,218  -7.30% 
 propane::virial      231,463 (+/- 18958)     251,724 (+/- 64689)            20,261   8.75% 
 water::energy_wolf   137,723 (+/- 5817)      131,462 (+/- 7529)             -6,261  -4.55% 
 water::forces_ewald  37,164,011 (+/- 90195)  34,768,217 (+/- 79851)     -2,395,794  -6.45% 
 water::forces_wolf   205,259 (+/- 10049)     223,179 (+/- 9897)             17,920   8.73% 
 water::virial_wolf   163,036 (+/- 14875)     157,260 (+/- 7941)             -5,776  -3.54% 



```



</p></details>

## Individual benchmarks

<details><summary>a4a8df4c Make force computations use mutable slices.</summary>
  <p>





```bash


running 4 tests
test argon::cache_move_particle ... bench:     327,171 ns/iter (+/- 2,877)
test argon::energy              ... bench:     312,417 ns/iter (+/- 37,336)
test argon::forces              ... bench:     393,919 ns/iter (+/- 2,371)
test argon::virial              ... bench:     405,909 ns/iter (+/- 3,583)

test result: ok. 0 passed; 0 failed; 0 ignored; 4 measured


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured


running 8 tests
test nacl::cache_move_particle_ewald ... bench:     656,441 ns/iter (+/- 1,090)
test nacl::cache_move_particle_wolf  ... bench:     167,559 ns/iter (+/- 1,524)
test nacl::energy_ewald              ... bench:   1,156,921 ns/iter (+/- 1,524)
test nacl::energy_wolf               ... bench:      98,585 ns/iter (+/- 4,610)
test nacl::forces_ewald              ... bench:  23,481,689 ns/iter (+/- 116,110)
test nacl::forces_wolf               ... bench:     149,863 ns/iter (+/- 1,596)
test nacl::virial_ewald              ... bench:   9,216,698 ns/iter (+/- 1,170,582)
test nacl::virial_wolf               ... bench:     112,928 ns/iter (+/- 772)

test result: ok. 0 passed; 0 failed; 0 ignored; 8 measured


running 5 tests
test propane::cache_move_all_rigid_molecules ... bench:   1,991,925 ns/iter (+/- 4,064)
test propane::cache_move_particles           ... bench:     509,915 ns/iter (+/- 2,622)
test propane::energy                         ... bench:     383,905 ns/iter (+/- 32,352)
test propane::forces                         ... bench:     480,464 ns/iter (+/- 34,835)
test propane::virial                         ... bench:     251,724 ns/iter (+/- 64,689)

test result: ok. 0 passed; 0 failed; 0 ignored; 5 measured


running 10 tests
test water::cache_move_all_rigid_molecules_ewald ... bench:  13,060,112 ns/iter (+/- 13,606)
test water::cache_move_all_rigid_molecules_wolf  ... bench:  11,607,241 ns/iter (+/- 420,651)
test water::cache_move_particles_ewald           ... bench:   1,368,830 ns/iter (+/- 7,914)
test water::cache_move_particles_wolf            ... bench:     762,936 ns/iter (+/- 112,616)
test water::energy_ewald                         ... bench:   1,877,087 ns/iter (+/- 2,148)
test water::energy_wolf                          ... bench:     131,462 ns/iter (+/- 7,529)
test water::forces_ewald                         ... bench:  34,768,217 ns/iter (+/- 79,851)
test water::forces_wolf                          ... bench:     223,179 ns/iter (+/- 9,897)
test water::virial_ewald                         ... bench:  13,070,197 ns/iter (+/- 495,643)
test water::virial_wolf                          ... bench:     157,260 ns/iter (+/- 7,941)

test result: ok. 0 passed; 0 failed; 0 ignored; 10 measured




```



</p></details>

<details><summary>3d72dc18 Typo Orthorombic => Orthorhombic</summary>
  <p>





```bash


running 4 tests
test argon::cache_move_particle ... bench:     326,005 ns/iter (+/- 2,718)
test argon::energy              ... bench:     312,334 ns/iter (+/- 4,654)
test argon::forces              ... bench:     394,827 ns/iter (+/- 2,332)
test argon::virial              ... bench:     404,528 ns/iter (+/- 17,683)

test result: ok. 0 passed; 0 failed; 0 ignored; 4 measured


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured


running 8 tests
test nacl::cache_move_particle_ewald ... bench:     654,262 ns/iter (+/- 1,037)
test nacl::cache_move_particle_wolf  ... bench:     165,983 ns/iter (+/- 2,360)
test nacl::energy_ewald              ... bench:   1,156,731 ns/iter (+/- 2,029)
test nacl::energy_wolf               ... bench:      98,623 ns/iter (+/- 1,240)
test nacl::forces_ewald              ... bench:  25,330,907 ns/iter (+/- 92,705)
test nacl::forces_wolf               ... bench:     151,865 ns/iter (+/- 6,462)
test nacl::virial_ewald              ... bench:   9,111,099 ns/iter (+/- 1,472,979)
test nacl::virial_wolf               ... bench:     112,486 ns/iter (+/- 827)

test result: ok. 0 passed; 0 failed; 0 ignored; 8 measured


running 5 tests
test propane::cache_move_all_rigid_molecules ... bench:   1,998,573 ns/iter (+/- 6,191)
test propane::cache_move_particles           ... bench:     516,541 ns/iter (+/- 3,674)
test propane::energy                         ... bench:     380,544 ns/iter (+/- 23,694)
test propane::forces                         ... bench:     478,047 ns/iter (+/- 22,221)
test propane::virial                         ... bench:     231,463 ns/iter (+/- 18,958)

test result: ok. 0 passed; 0 failed; 0 ignored; 5 measured


running 10 tests
test water::cache_move_all_rigid_molecules_ewald ... bench:  13,167,571 ns/iter (+/- 13,346)
test water::cache_move_all_rigid_molecules_wolf  ... bench:  11,735,771 ns/iter (+/- 386,027)
test water::cache_move_particles_ewald           ... bench:   1,390,598 ns/iter (+/- 37,725)
test water::cache_move_particles_wolf            ... bench:     765,670 ns/iter (+/- 42,322)
test water::energy_ewald                         ... bench:   1,890,358 ns/iter (+/- 1,380)
test water::energy_wolf                          ... bench:     137,723 ns/iter (+/- 5,817)
test water::forces_ewald                         ... bench:  37,164,011 ns/iter (+/- 90,195)
test water::forces_wolf                          ... bench:     205,259 ns/iter (+/- 10,049)
test water::virial_ewald                         ... bench:  13,011,808 ns/iter (+/- 657,861)
test water::virial_wolf                          ... bench:     163,036 ns/iter (+/- 14,875)

test result: ok. 0 passed; 0 failed; 0 ignored; 10 measured




```



</p></details>

<br>**CPU**: Intel(R) Xeon(R) CPU E5-2660 0 @ 2.20GHz

**Comment by Luthaf** [comment]
The build failed with this output:

```
     Running target/release/deps/md_nacl-2a92d69993c8fac2
running 4 tests
test ewald::energy ... ok
test ewald::constant_energy ... ok
test wolf::constant_energy ... ok
test wolf::anisotropic_berendsen has been running for over 60 seconds

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```

I already had this problem, and restarting the build made it go away, so I am restarting this one too !

**Comment by antoinewdg** [comment]
Rebased :)

**Comment by Luthaf** [comment]
The coverage drop seems to be linked to an update in kcov. The missing lines are mainly closing braces and other non-code lines.

So I am OK with merging this, with an additional commit to update (again ...) the codecov/target value.

I hope we get code coverage from rustc soon, it should improve these issues 😄.

**Comment by Luthaf** [comment]
The coverage failure makes no sense, as it does not appear on other PR. Could you rebase it once more, just to check?

**Comment by antoinewdg** [comment]
Done

**Comment by antoinewdg** [comment]
This is getting annoying (the previous build also errored). I made a new rebase because the previous code could not be merged, let's see what it does.

**Comment by antoinewdg** [comment]
Do we just merge anyway ?

**Comment by Luthaf** [comment]
Yeah, let's do it. We'll see what happens with code coverage.

**Pull request merged**